### PR TITLE
Add some crate level documentation

### PR DESCRIPTION
**Pull request by nlordell**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,87 @@
...
+//! use std::time::Duration;
+//! use web3::api::Web3;
+//! use web3::types::*;
+//!
+//! // this proc macro generates a `MyContract` type with type-safe bindings to
+//! // contract functions
+//! ethcontract::contract!("path/to/MyContract.json");
+//!
+//! // create a web3 instance as usual
+//! let transport = ...;
```
Comment (bh2smith) [inline src/lib.rs:47]: Is this "transport" information less important? the ... makes me wonder what I am missing!

**Review by bh2smith — APPROVED** [review]
Very nice looking documentation! 🎸

File: src/lib.rs (hunk 1 above)
Comment (nlordell) [reply src/lib.rs:47]: @bh2smith it's `web3` related so I omitted it from this example. Its basically how you want to connect to a node.

File: src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,87 @@
 #![deny(missing_docs)]
```
Comment (fleupold) [inline src/lib.rs:1]: Will we be able to get rid of this eventually?

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FdsdVyKkSqccEzoPufX%2Fgiphy.gif&token=documentation" data-gitmeme-token="documentation"></a>

File: src/lib.rs (hunk 2 above)
Comment (nlordell) [reply src/lib.rs:1]: No, it's so if we ever add a new public facing type, const, function, or module to `ethcontract-rs` but forget to document it, we get build errors.

Comment (fleupold) [reply src/lib.rs:1]: sorry, I read `allow(missing_docs)`, deny is awesome!

Comment (bh2smith) [reply src/lib.rs:1]: Denied!

**Pull request merged**